integrations/symfony-bundle.rst

```diff
@@ -0,0 +1,176 @@
+
+.. code-block:: yaml
+
+    httplug:
+        clients:
+            acme_client: # This is the name of the client
+            factory: 'httplug.factory.guzzle6'
+
+        main_alias:
+            client: httplug.client.default
+            message_factory: httplug.message_factory.default
+            uri_factory: httplug.uri_factory.default
+            stream_factory: httplug.stream_factory.default
+        classes:
+            # uses discovery if not specified
+            client: ~
+            message_factory: ~
+            uri_factory: ~
+            stream_factory: ~
+
+
+Configuration without auto discovery
+````````````````````````````````````
+
+By default we use Puli to auto discover factories. If you do not want to use auto discovery you could use the following configuration (Guzzle):
+
+.. code-block:: yaml
+
+    httplug:
+        classes:
+            client: Http\Adapter\Guzzle6\Client
+            message_factory: Http\Message\MessageFactory\GuzzleMessageFactory
+            uri_factory: Http\Message\UriFactory\GuzzleUriFactory
+            stream_factory: Http\Message\StreamFactory\GuzzleStreamFactory
+
+
+Configure your client
+`````````````````````
+
+You can configure your clients with some good default options. The clients are later registered as services.
+
+.. code-block:: yaml
+
+    httplug:
+        clients:
+            my_guzzle5:
+                factory: 'httplug.factory.guzzle5'
+                config:
+                    # These options are given to Guzzle without validation.
+                    defaults:
+                        base_uri: 'http://google.se/'
+                        verify_ssl: false
+                        timeout: 4
+                        headers:
+                            Content-Type: 'application/json'
+            acme:
+                factory: 'httplug.factory.guzzle6'
+                config:
+                    base_uri: 'http://google.se/'
+
+.. code-block:: php
+
+    $httpClient = $this->container->get('httplug.client.my_guzzle5');
+    $httpClient = $this->container->get('httplug.client.acme');
+
+
+Plugins
+```````
+
+You can configure the clients with plugins.
+
+.. code-block:: yaml
+
+    // services.yml
+    acme_plugin:
+          class: Acme\Plugin\MyCustonPlugin
+          arguments: ["%api_key%"]
+
+.. code-block:: yaml
+
+    // config.yml
+    httpug:
+        plugins:
+            cache:
+                cache_pool: 'my_cache_pool'
+        clients:
+            acme:
+                factory: 'httplug.factory.guzzle6'
+                plugins: ['acme_plugin', 'httplug.plugin.cache', ''httplug.plugin.retry']
+                config:
+                    base_uri: 'http://google.se/'
+
+
+Authentication
+``````````````
+
```

Comment:
explain what this is and link to the authentication documentation for background